### PR TITLE
seccomp: implement syscall continuation

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

File: lxd/api_1.0.go [hunk 1]
```diff
@@ -205,11 +205,12 @@ func api10Get(d *Daemon, r *http.Request) response.Response {
...
-		"netnsid_getifaddrs": fmt.Sprintf("%v", d.os.NetnsGetifaddrs),
-		"uevent_injection":   fmt.Sprintf("%v", d.os.UeventInjection),
-		"unpriv_fscaps":      fmt.Sprintf("%v", d.os.VFS3Fscaps),
-		"seccomp_listener":   fmt.Sprintf("%v", d.os.SeccompListener),
-		"shiftfs":            fmt.Sprintf("%v", d.os.Shiftfs),
+		"netnsid_getifaddrs":                fmt.Sprintf("%v", d.os.NetnsGetifaddrs),
+		"uevent_injection":                  fmt.Sprintf("%v", d.os.UeventInjection),
+		"unpriv_fscaps":                     fmt.Sprintf("%v", d.os.VFS3Fscaps),
+		"seccomp_listener":                  fmt.Sprintf("%v", d.os.SeccompListener),
+		"seccomp_listener_continue_syscall": fmt.Sprintf("%v", d.os.SeccompListenerContinue),
```
Comment (stgraber) [inline lxd/api_1.0.go]: How about we just call it `seccomp_listener_continue` to keep things nice and short?

File: lxd/seccomp/seccomp.go [hunk 2]
```diff
@@ -926,16 +901,28 @@ func (s *Server) HandleMknodSyscall(c Instance, siov *Iovec) int {
...
 
-	siov.resp.error = C.device_allowed(C.dev_t(siov.req.data.args[2]), C.mode_t(siov.req.data.args[1]))
-	if siov.resp.error != 0 {
+	if C.device_allowed(C.dev_t(siov.req.data.args[2]), C.mode_t(siov.req.data.args[1])) < 0 {
 		logger.Debugf("Device not allowed")
+		if s.s.OS.SeccompListenerContinue {
+			logger.Debugf("Continuing mknod syscall")
+			C.seccomp_notify_update_response(siov.resp, 0, C.uint32_t(seccompUserNotifFlagContinue))
+			return 0
+		}
```
Comment (stgraber) [inline lxd/seccomp/seccomp.go]: Can you reshuffle things a little so we don't en up logging two messages per syscall in the continue case?

Comment (stgraber) [reply lxd/seccomp/seccomp.go]: And do the same to all the other cases below?

File: lxd/api_1.0.go (hunk 1 above)
Comment (brauner) [reply lxd/api_1.0.go]: done

File: lxd/seccomp/seccomp.go (hunk 2 above)
Comment (brauner) [reply lxd/seccomp/seccomp.go]: completely reworked

**Pull request merged**